site/public/js/user-profile.js

```diff
@@ -273,4 +273,10 @@ $(document).ready(function() {
     // Set time zone drop down boxes to the user's time zone (only after other JS has finished loading)
     let user_time_zone =  $('#time_zone_selector_label').data('user_time_zone');
     $('[value="' + user_time_zone + '"]').prop('selected', true);
+    $('#user-image-button').bind('change', function() {
+        if ((this.files[0].size/1048576)>20.0){
+            alert("Selected file's size exceeds 20 MB");
+            $('#user-image-button').val('');
+        }
+    });
```

Comment:
Is this being enforced on the server side somewhere besides the default upload size settings?

Reply:
I also wonder whether 20 MB is an appropriate limit for a profile picture.  If everyone in a 300-person class uploaded a 20 MB file, going to the page which lists student profiles would result in a very very large download.  Would a more appropriate limit be 2MB?  Maybe 5MB?  It would be better to have an even lower (<1MB) limit in my opinion, but it would likely require most students to compress their images somewhat which may be desirable.

Reply:
I agree, animated GIFs don't seem good in educational settings. I tested with some animated ones and they lost their animated nature, but they also got kind of corrupted. If a static GIF is uploaded it shows up fine. 

Image dimensions are being resized server side, due to this no image will probably have 20MB size. I agree that 20MB is a bit large, I relied on server-side resize for compression. I think having to compress image yourself can be a hassle and deteriorates user experience. It should definitely be decreased but I'm also not sure about the exact number.